### ISSUE TITLE
Laravel mix not interpreting Vue components watchers as intended

### ISSUE THREAD

**Issue by vTripes** [JeffreyWay/laravel-mix#262]
- Laravel Version: 5.4.6
- Vue Version: 2.1.10
- Mix Version: 0.5.16

### Description:
Since Laravel 5.4, it is recommended to use Laravel mix (config layer of webpack) to compile assets, including Vue files.
However, it seems to be interpreting code quite badly since it make components that were working dysfunctional.
It seems that setting a preset function as a component watcher's handler changes the scope and then translate `this.prop` in the handler function's parameters to `undefined.prop`.
As for information, I was using Gulp to transpile assets before Laravel 5.4.

### Steps To Reproduce:
- Create a vue single file component with these required :
```
props: {
  myObjectProp: {
    type: Object
  }
},
computed: {
  debounceTime () {
    return (condition == expected) ? 500 : 0
  }
},
watch: {
  myObjectProp: {
    handler: _.debounce(function (val) {
      // Do some stuff here
    }, this.debounceTime),
    deep: true
  }
}
```
- Then compile by running `npm run dev`

**Comment by JeffreyWay** [comment]
Well you have a function there, so that should change the value of `this`, right? What happens if you change it to:

```js
watch: {
  myObjectProp: {
    handler: _.debounce(val => {
      // Do some stuff here
    }, this.debounceTime),
    deep: true
  }
}
```

**Comment by vTripes** [comment]
It doesn't change anything, and it shouldn't change the value of this because if you see, `this.debounceTime` is outside of the callback closure, meaning that it is passed as a parameter.
And if I'm not mistaken, passing `this.parameter` is something that isn't forbidden at all.
The handler is supposed to be fired everytime the value of `myObjectProp` changes, but this is what happen if I do this:
```JavaScript
watch: {
  myObjectProp: {
    handler: function(val) {
      _.debounce(function () {
        // Do some stuff here
      }, this.debounceTime)
    },
    deep: true
  }
}
```
First, this solution is not concise because of the anonym function wrapping lodash's debounce function.
Then, it is not really fixed because the handler will be fired only when the Vue render updates. (meaning that for a datatable in my case, you have to change the sorting to see the filter applied)

**Comment by JeffreyWay** [comment]
Okay, I misunderstood you. 

I just recreated your example, and then checked the compiled code. I see no reference to undefined.debounceTime, and everything seems to be working properly. 

https://www.dropbox.com/s/moh8rz7lhl739dj/Screenshot%202017-02-01%2011.34.04.png?dl=0

**Comment by vTripes** [comment]
We found out what went wrong with the compilation.
Acutally, I did try to recreate a reproduction to provide, and was unable to recreate the behavior described above. This way, we found out that there was a file in the application's root folder named `.babelrc` containing these lines:
```JSON
{
    "presets": ["es2015"]
}
```
And this was the source of the problem. If this is in the project, the compilation goes wrong and code doesn't work as inteded anymore.
So if people have compilation problem, but the code seems fine, they should take a look to see if they have these lines forcing the code to not go to ES5.

Thank you for taking the time to answer me and try to figure out what was the problem.
